webpack.config.js

```diff
@@ -150,6 +150,171 @@ const GOOGLESITEKIT_VERSION = googleSiteKitVersion
+// } );
+
+const gutenbergEntryPoints = {
+	'googlesitekit-idea-hub-notice':
+		'./assets/js/googlesitekit-idea-hub-notice.js',
+};
+
+const customExternalsBundle = (
+	mode,
+	entryPoints,
+	externalDependencies,
+	analyze
+) => {
+	const rules = createRules( mode );
+
+	return {
+		entry: entryPoints,
+		externals: externalDependencies,
+		output: {
+			filename:
+				mode === 'production' ? '[name]-[contenthash].js' : '[name].js',
+			path: path.join( __dirname, 'dist/assets/js' ),
+			chunkFilename:
+				mode === 'production' ? '[name]-[chunkhash].js' : '[name].js',
+			publicPath: '',
+			/*
+                 If multiple webpack runtimes (from different compilations) are used on the
+                 same webpage, there is a risk of conflicts of on-demand chunks in the global
+                 namespace.
+                 See: https://webpack.js.org/configuration/output/#outputjsonpfunction.
+             */
+			jsonpFunction: '__googlesitekit_webpackJsonp',
+		},
+		performance: {
+			maxEntrypointSize: 175000,
+		},
+		module: {
+			rules: [ ...rules ],
+		},
+		plugins: [
+			new WebpackBar( {
+				name: 'Gutenberg Entry Points',
+				color: '#ffc0cb',
+			} ),
+			new CircularDependencyPlugin( {
+				exclude: /node_modules/,
+				failOnError: true,
+				allowAsyncCycles: false,
+				cwd: process.cwd(),
+			} ),
+			new CreateFileWebpack( {
+				path: './dist',
+				fileName: 'config.json',
+				content: JSON.stringify( {
+					buildMode: mode,
+					features,
+				} ),
+			} ),
+			new ManifestPlugin( {
+				...manifestArgs,
+				filter( file ) {
+					return ( file.name || '' ).match( /\.js$/ );
+				},
+			} ),
+			new DefinePlugin( {
+				'global.GOOGLESITEKIT_VERSION': JSON.stringify(
+					GOOGLESITEKIT_VERSION
+				),
+			} ),
+			new ESLintPlugin( {
+				emitError: true,
+				emitWarning: true,
+				failOnError: true,
+			} ),
+			...( analyze ? [ new BundleAnalyzerPlugin() ] : [] ),
+		],
+		optimization: {
+			minimizer: [
+				new TerserPlugin( {
+					parallel: true,
+					sourceMap: mode !== 'production',
+					cache: true,
+					terserOptions: {
+						// We preserve function names that start with capital letters as
+						// they're _likely_ component names, and these are useful to have
+						// in tracebacks and error messages.
+						keep_fnames: /__|_x|_n|_nx|sprintf|^[A-Z].+$/,
+						output: {
+							comments: /translators:/i,
+						},
+					},
+					extractComments: false,
+				} ),
+			],
+			/*
+                 The runtimeChunk value 'single' creates a runtime file to be shared for all generated chunks.
+                 Without this, imported modules are initialized for each runtime chunk separately which
+                 results in duplicate module initialization when a shared module is imported by separate entries
+                 on the same page.
+                 See: https://v4.webpack.js.org/configuration/optimization/#optimizationruntimechunk
+             */
+			// runtimeChunk: 'single',
```

Comment:
```suggestion
```suggestion
		// The runtimeChunk value 'single' creates a runtime file to be shared for all generated chunks.
		// Without this, imported modules are initialized for each runtime chunk separately which
		// results in duplicate module initialization when a shared module is imported by separate entries
		// on the same page.
		// See: https://v4.webpack.js.org/configuration/optimization/#optimizationruntimechunk
		runtimeChunk: 'single',
```

Why is this commented out? I assume with the explanation above this was intended to stay? Or should this all be removed?